### PR TITLE
Make search for graphics and sound folders case insensitive and unify it between Android and desktop version (fixes #489)

### LINKED ISSUE #489: Support upper/lower case file names for graphics.
Under Linux several things are not rendered (hidden?):
broken settlers rendering
This is the corresponding log:

```
$ java -jar JSettlers.jar 
No default command line options file found.
Options loaded: {}
Source: (file:/***/JSettlers.jar <no signer certificates>)
Could not find/load graphic file 01
Could not find/load graphic file 11
java.lang.NullPointerException
    at jsettlers.graphics.map.draw.ImagePreloadTask.run(ImagePreloadTask.java:121)
    at java.lang.Thread.run(Thread.java:745)
Could not find/load graphic file 12
java.lang.NullPointerException
    at jsettlers.graphics.map.draw.ImagePreloadTask.run(ImagePreloadTask.java:189)
    at java.lang.Thread.run(Thread.java:745)
Could not find/load graphic file 03
Could not find/load graphic file 03
Warning: Could not load LocaleSuffix [locale=en_US, useCountry=true]. Falling back to next file.
Loading Sequence number 17
Loading Sequence number 18
```

Under Windows, with the same data files, it works fine.

### PR DESCRIPTION
**Pull request by rpolzer**
resolves issue #489: gfx files may have arbitrary upper or lower case file names

### CODE REVIEW

**Comment by rpolzer** [comment]
seems to resolve issue #310, too

**Comment by andreas-eberle** [comment]
@rpolzer: Many thanks for all your PRs! I'll review them as soon as possible! But I first want to finish the #532. If you have any questions or so, feel free to join our discord https://discord.gg/2hVV4u6.

File: jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/ImageProvider.java [hunk 1]
```diff
@@ -300,9 +300,10 @@ public void invalidateAll() {
 
 	private File findFileInPaths(String fileName) {
 		for (File path : this.lookupPaths) {
-			File searched = new File(path, fileName);
-			if (searched.isFile() && searched.canRead()) {
-				return searched;
+			for (File f : path.listFiles()) {
+				if (f.getName().equalsIgnoreCase(fileName) && f.isFile() && f.canRead()) {
```
Comment (andreas-eberle) [inline jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/ImageProvider.java]: The condition of this should better be
`f.isFile() && f.canRead() && fileName.equalsIgnoreCase(f.getName()) `
This way, we can prevent a potential NullPointer if f.getName returns null and the simpler checks are done first.

File: jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/ImageProvider.java [hunk 2]
```diff
@@ -300,9 +300,10 @@ public void invalidateAll() {
 
 	private File findFileInPaths(String fileName) {
 		for (File path : this.lookupPaths) {
-			File searched = new File(path, fileName);
-			if (searched.isFile() && searched.canRead()) {
-				return searched;
+			for (File f : path.listFiles()) {
```
Comment (andreas-eberle) [inline jsettlers.graphics/src/main/java/jsettlers/graphics/map/draw/ImageProvider.java]: Please don't use single letter variable names for this. Calling it currentFile is more obvious (I know, here isn't much code around, but we want to keep a clean code in the project ;) )

File: jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java [hunk 3]
```diff
@@ -64,13 +62,16 @@ public static SettlersFolderInfo checkSettlersFolder(String settlersBaseFolder)
 
 		for (File f : files) {
 			if (f.getName().equalsIgnoreCase("snd")) {
-				File testSndFile = new File(f, "Siedler3_00.dat");
-				if (testSndFile.exists()) {
-					sndFolder = f;
+				for (File g : f.listFiles()) {
```
Comment (andreas-eberle) [inline jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java]: Same as before, please no one letter variable names (except for index i in for loop)

File: jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java [hunk 4]
```diff
@@ -64,13 +62,16 @@ public static SettlersFolderInfo checkSettlersFolder(String settlersBaseFolder)
...
+					g.getName();
```
Comment (andreas-eberle) [inline jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java]: Did you mean to have this line here?

File: jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java [hunk 5]
```diff
@@ -64,13 +62,16 @@ public static SettlersFolderInfo checkSettlersFolder(String settlersBaseFolder)
...
+					if (g.getName().equalsIgnoreCase("siedler3_00.dat")) {
```
Comment (andreas-eberle) [inline jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java]: Again, switch the argument and the object to 
`"siedler3_00.dat".equalsIgnoreCase(g.getName())`

File: jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java [hunk 6]
```diff
@@ -64,13 +62,16 @@ public static SettlersFolderInfo checkSettlersFolder(String settlersBaseFolder)
...
+						sndFolder = f;
+					}
 				}
 			} else if (f.getName().equalsIgnoreCase("gfx")) {
-				for (DatFileType t : DatFileType.values()) {
-					if (new File(f, "siedler3_00" + t.getFileSuffix()).exists()) {
+				for (File g : f.listFiles()) {
+					String a = g.getName();
+					if (a.substring(0, a.indexOf('.')).equalsIgnoreCase("siedler3_00")) {
```
Comment (andreas-eberle) [inline jsettlers.main.swing/src/main/java/jsettlers/main/swing/resources/SettlersFolderChecker.java]: Same as above.

**Comment by rpolzer** [comment]
have a look...

**Comment by andreas-eberle** [comment]
The changes look good. One more thing however: While reviewing one of the other PRs, I saw that the sound file is also searched in SoundManager:221. So the case insensitve check would also be needed there. However, I think it would be best to consolidate these codes into the SettlersFolderChecker class to have it all in one place. Can you do this, @rpolzer?

**Comment by rpolzer** [comment]
I tried this, but the result is either a gradle error
`Error:(31, 38) error: package jsettlers.main.swing.resources does not exist`
`Error:(192, 18) error: cannot find symbol variable SettlersFolderChecker`
or a circular dependency
`Error:Circular reference between projects: :jsettlers.main.swing -> :jsettlers.graphics -> :jsettlers.main.swing`
because the module graphics (which needs the file lookup) can not depend on the module main.swing (where I tried to put the code).

So the function
`File findFileInPaths(String fileName, List<File> lookupPaths)`
should reside somewhere in the module graphics and not in the class SettlersFolderChecker.

**Comment by andreas-eberle** [comment]
I'll try to sort this out.

**Comment by andreas-eberle** [comment]
@rpolzer: I had to rework and sort out a lot of old messy code... However, now I think it should be ok and the same code is running on the desktop and the Android version. Can you have a look again to double-check?

**Comment by rpolzer** [comment]
It is still not possible to have findFileInPath inside the class SettlersFolderChecker.
In my opinion, this is not necessary and I would leave it where it is.
I tested this branch on Linux: changing of graphics file names upper and lower case letters is OK.

**Pull request merged**